lib/handlers/hard-quote/handler.ts

```diff
@@ -240,3 +262,64 @@ function getDecayEndTime(chainId: number, startTime: number): number {
       return startTime + 30; // 30 seconds
   }
 }
+
+function createHardQuoteResponse(
+  request: HardQuoteRequest,
+  order: CosignedV2DutchOrder | CosignedV3DutchOrder
+): V2HardQuoteResponse | V3HardQuoteResponse {
+  if (order instanceof CosignedV2DutchOrder) {
+    return new V2HardQuoteResponse(request, order);
+  } else if (order instanceof CosignedV3DutchOrder) {
+    return new V3HardQuoteResponse(request, order);
+  }
+  throw new Error('Unsupported order type');
+}
+
+async function createCosignedOrder(
+  cosigner: KmsSigner,
+  request: HardQuoteRequest,
+  cosignerData: CosignerData | V3CosignerData,
+): Promise<CosignedV2DutchOrder | CosignedV3DutchOrder> {
+  if (request.order instanceof UnsignedV2DutchOrder) {
+    const v2CosignerData = cosignerData as CosignerData;
+    const cosignature = await cosigner.signDigest(request.order.cosignatureHash(v2CosignerData));
+    return CosignedV2DutchOrder.fromUnsignedOrder(request.order, v2CosignerData, cosignature);
+  } else if (request.order instanceof UnsignedV3DutchOrder) {
+    const v3CosignerData = cosignerData as V3CosignerData;
+    const cosignature = await cosigner.signDigest(request.order.cosignatureHash(v3CosignerData));
+    return CosignedV3DutchOrder.fromUnsignedOrder(request.order, v3CosignerData, cosignature);
+  } else {
+    throw new Error('Unsupported order type');
+  }
+}
+
+function getDefaultV2CosignerData(request: HardQuoteRequest): CosignerData {
+  const decayStartTime = getDecayStartTime(request.tokenInChainId);
+  const filler = ethers.constants.AddressZero;
+  let inputOverride = BigNumber.from(0);
+  const outputOverrides = request.order.info.outputs.map(() => BigNumber.from(0));
+  if (request.type === TradeType.EXACT_INPUT) {
+    outputOverrides[0] = request.totalOutputAmountStart;
+  } else {
+    inputOverride = request.totalInputAmountStart;
+  }
+
+  return {
+    decayStartTime: decayStartTime,
+    decayEndTime: getDecayEndTime(request.tokenInChainId, decayStartTime),
+    exclusiveFiller: filler,
+    exclusivityOverrideBps: DEFAULT_EXCLUSIVITY_OVERRIDE_BPS,
+    inputOverride: inputOverride,
+    outputOverrides: outputOverrides,
+  };
+}
+
+function getDefaultV3CosignerData(request: HardQuoteRequest): V3CosignerData {
+  return {
+    decayStartBlock: 1,
```

Comment:
This is the absolute block number (not relative)

Reply:
Did some thinking on this, I think we'll need to add something to injector that we can use the grab the L2 block and then fill this field with `L2Block + V3_BLOCK_BUFFER`

Reply:
Addressed in latest commit